src/main/java/com/ericsson/eiffel/remrem/protocol/MsgService.java

```diff
@@ -30,6 +30,18 @@
      * @return the generated and validate Eiffel messages as json String
      */
     String generateMsg(String msgType, JsonObject jsonMessage);
+    
+    /**
+     * Generate takes a partial message in JSON format, 
+     * validates it and adds mandatory fields before outputting a complete, valid Eiffel message.
+     * lenientValidation will perform the only mandatory field validation and non-mandatory validation failures will place in Eiffel message as a new property(remremGenerateFailures)
+     * 
+     * @param msgType
+     * @param jsonMessage
+     * @param lenientValidation
+     * @return the generated and validate Eiffel messages as json String
+     */
+    String generateMsg(String msgType, JsonObject jsonMessage, Boolean lenientValidation);

```

Comment:
final variables ? why Boolean, can we not have boolean

Reply:
final good to have in implementation. Boolean(Wrappers) to get all the fancy Java features